query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -85,12 +121,36 @@ class AvgOverTimeFunction extends RangeFunction {
   }
 }
 
-class StdDevOverTimeFunction extends RangeFunction {
+abstract class AvgOverTimeChunkedFunction(var sum: Double = 0d, var count: Int = 0) extends ChunkedRangeFunction {
+  override final def reset(): Unit = { sum = 0d; count = 0 }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, if (count > 0) sum/count else 0d)
+  }
+}
+
+class AvgOverTimeChunkedFunctionD extends AvgOverTimeChunkedFunction() with ChunkedDoubleRangeFunction {
+  final def addTimeDoubleChunks(doubleVect: BinaryVector.BinaryVectorPtr,
+                                doubleReader: bv.DoubleVectorDataReader,
+                                startRowNum: Int,
+                                endRowNum: Int): Unit = {
+    sum += doubleReader.sum(doubleVect, startRowNum, endRowNum)
+    count += (endRowNum - startRowNum + 1)
+  }
+}
 
-  var sum = 0d
-  var count = 0
-  var squaredSum = 0d
+class AvgOverTimeChunkedFunctionL extends AvgOverTimeChunkedFunction() with ChunkedLongRangeFunction {
+  final def addTimeLongChunks(longVect: BinaryVector.BinaryVectorPtr,
+                              longReader: bv.LongVectorDataReader,
+                              startRowNum: Int,
+                              endRowNum: Int): Unit = {
+    sum += longReader.sum(longVect, startRowNum, endRowNum)
+    count += (endRowNum - startRowNum + 1)
```

Comment:
End-Of-Time-Series marker needs to be allowed and handled for both sum and count

Reply:
Again, we will punt on it for Long type data as there is no way the end of time series marker is generated for long columns today (since Prom doesn't output Long type data)